src/main/java/uk/ac/ebi/eva/pipeline/EvaPipelineJobLauncherCommandLineRunner.java

```diff
@@ -0,0 +1,277 @@
+        ApplicationEventPublisherAware, ExitCodeGenerator {
+
+    private static final Logger logger = LoggerFactory.getLogger(EvaPipelineJobLauncherCommandLineRunner.class);
+
+    public static final String SPRING_BATCH_JOB_NAME_PROPERTY = "spring.batch.job.names";
+
+    public static final int EXIT_WITHOUT_ERRORS = 0;
+
+    private static final int EXIT_WITH_ERRORS = 1;
+
+    @Value("${" + SPRING_BATCH_JOB_NAME_PROPERTY + ":#{null}}")
+    private String jobName;
+
+    @Value("${" + JobParametersNames.PROPERTY_FILE_PROPERTY + ":#{null}}")
+    private String propertyFilePath;
+
+    @Value("${" + JobParametersNames.RESTART_PROPERTY + ":false}")
+    private boolean restartPreviousExecution;
+
+    private Collection<Job> jobs;
+
+    private JobRepository jobRepository;
+
+    private JobRegistry jobRegistry;
+
+    private JobParametersConverter converter;
+
+    @Autowired
+    private JobExecutionApplicationListener jobExecutionApplicationListener;
+
+    public EvaPipelineJobLauncherCommandLineRunner(JobLauncher jobLauncher, JobExplorer jobExplorer,
+                                                   JobRepository jobRepository) {
+        super(jobLauncher, jobExplorer);
+        jobs = Collections.emptySet();
+        this.jobRepository = jobRepository;
+        converter = new DefaultJobParametersConverter();
+    }
+
+    @Autowired(required = false)
+    public void setJobRegistry(JobRegistry jobRegistry) {
+        this.jobRegistry = jobRegistry;
+    }
+
+    @Autowired(required = false)
+    public void setJobParametersConverter(JobParametersConverter converter) {
+        this.converter = converter;
+    }
+
+    @Autowired(required = false)
+    public void setJobs(Collection<Job> jobs) {
+        this.jobs = jobs;
+    }
+
+    public void setJobName(String jobName) {
+        this.jobName = jobName;
+    }
+
+    public void setPropertyFilePath(String propertyFilePath) {
+        this.propertyFilePath = propertyFilePath;
+    }
+
+    @Override
+    public int getExitCode() {
+        if (jobExecutionApplicationListener.isJobExecutionComplete()) {
+            return EXIT_WITHOUT_ERRORS;
+        } else {
+            return EXIT_WITH_ERRORS;
+        }
+    }
+
+    @Override
+    public void run(String... args) throws JobExecutionException {
+        try {
+            JobParameters jobParameters = getJobParameters(args);
+            if (restartPreviousExecution) {
+                restartPreviousJobExecution(jobParameters);
+            }
+            launchJob(jobParameters);
+        } catch (NoJobToExecuteException | NoParametersHaveBeenPassedException | UnexpectedFileEncodingException
+                | FileNotFoundException | UnexpectedErrorReadingFileException | NoPreviousJobExecutionException
+                | NotValidParameterFormatException | UnknownJobException e) {
+            logger.error(e.getMessage());
+            logger.trace("Error trace", e);
```

Comment:
Because they already return a good indicative message that shouldn't require more information to identify what are you doing wrong.